loky/backend/_thread_pool_limiters.py

```diff
@@ -0,0 +1,481 @@
+    # happen, by dynamically reconfiguring the inner OpenMP runtime to
+    # temporarily disable it while under the scope of the outer OpenMP parallel
+    # section.
+    os.environ.setdefault("KMP_DUPLICATE_LIB_OK", "True")
+
+
+def openmp_num_threads(n_threads=None):
+    """Determine the effective number of threads used for parallel OpenMP calls
+
+    For n_threads=None, returns openmp.omp_get_max_threads().
+    For n_threads > 0, use this as the maximal number of threads for parallel
+    OpenMP calls and for n_threads < 0, use the maximal number of threads minus
+    - n_threads + 1.
+    Raise a ValueError for n_threads=0.
+    """
+    if n_threads == 0:
+        raise ValueError("n_threads=0 is invalid")
+    elif n_threads < 0:
+        return max(1, cpu_count() + n_threads + 1)
+    else:
+        return n_threads
+
+
+# Structure to cast the info on dynamically loaded library. See
+# https://linux.die.net/man/3/dl_iterate_phdr for more details.
+UINT_SYSTEM = ctypes.c_uint64 if sys.maxsize > 2**32 else ctypes.c_uint32
+UINT_HALF_SYSTEM = ctypes.c_uint32 if sys.maxsize > 2**32 else ctypes.c_uint16
+
+
+# Constants of the module
+NOT_ACCESSIBLE = 'Not accessible'
+
+
+class dl_phdr_info(ctypes.Structure):
+    _fields_ = [
+        ("dlpi_addr",  UINT_SYSTEM),       # Base address of object
+        ("dlpi_name",  ctypes.c_char_p),   # path to the library
+        ("dlpi_phdr",  ctypes.c_void_p),   # pointer on dlpi_headers
+        ("dlpi_phnum",  UINT_HALF_SYSTEM)  # number of element in dlpi_phdr
+        ]
+
+
+class _CLibsWrapper:
+    # Wrapper around classic C-libraries for scientific computations to set and
+    # get the maximum number of threads they are allowed to used for inner
+    # parallelism.
+
+    # supported API
+    SUPPORTED_API = {
+        "omp": {
+            "set": "omp_set_num_threads",
+            "get": "omp_get_max_threads"},
+        "openblas": {
+            "set": "openblas_set_num_threads",
+            "get": "openblas_get_num_threads"},
+        "mkl": {
+            "set": "MKL_Set_Num_Threads",
+            "get": "MKL_Get_Max_Threads"},
+    }
+
+    # Supported C-libraries for this wrapper, index with their name. The items
+    # hold the name of the library file and the functions to call.
+    SUPPORTED_CLIBS = {
+        "openmp_intel": {
+            "pattern": "libiomp",
+            "api": "omp"},
+        "openmp_gnu": {
+            "pattern": "libgomp",
+            "api": "omp"},
+        "openmp_llvm": {
+            "pattern": "libomp",
+            "api": "omp"},
+        "openmp_win32": {
+            "pattern": "vcomp",
+            "api": "omp"},
+        "openblas": {
+            "pattern": "libopenblas",
+            "api": "openblas"},
+        "mkl": {
+            "pattern": "libmkl_rt",
+            "api": "mkl"},
+        "mkl_win32": {
+            "pattern": "mkl_rt",
+            "api": "mkl"},
```

Comment:
To me the `"api"` for MKL and openblas should be `"blas"`.

Reply:
Also, maybe we should collapse "mkl" and "mkl_win32" into a single entry and allow for multiple patterns:

```python
        "mkl": {
            "pattern": ("libmkl_rt", "mkl_rt"),
            "api": "blas"},
```

Those two entries are fundamentally the same library, it's just the filename / packaging that differs depending on the OS.

Reply:
And `"pattern"` should be renamed to `"filename_prefix"` to be more explicit.